src/utils/HistoryStore.ts

```diff
@@ -0,0 +1,21 @@
+import { history } from 'coveo.analytics';
+import { CookieStorage, getAvailableStorage, NullStorage } from 'coveo.analytics/dist/storage';
+
+export function buildHistoryStore() {
+  const historyStore = buildCookieHistoryStore();
+  (historyStore as any).store = getAvailableStorage();
+
+  return historyStore;
+}
+
+export function buildNullHistoryStore() {
+  const historyStore = buildCookieHistoryStore();
+  (historyStore as any).store = new NullStorage();
+
+  return historyStore;
+}
+
+function buildCookieHistoryStore() {
+  const cookieStorage = new CookieStorage();
+  return new history.HistoryStore(cookieStorage);
```

Comment:
Before you changes, there was no parameter `new history.HistoryStore()`, so why are you using `cookieStorage` as a parameter now?

Reply:
Passing CookieStorage to the constructor prevents the class in `coveo.analytics` from writing to cookies. I initially adjusted the logic [here](https://github.com/coveo/coveo.analytics.js/pull/216/files), but encountered errors when trying to upgrade the package to use the changes.

Reply:
It's not totally clear for me with this PR, but this is important:  If analytics is enabled, then actions history will still be kept in local storage, right ? (And not cookies).

If yes, then :shipit:  !

Reply:
Yes, once the class is instantiated, the storage mechanism is adjusted on line 6 using the `getAvailableStorage` utility. The function prioritizes localstorage over cookies, and cookies over session storage. It's implementation is available here: https://github.com/coveo/coveo.analytics.js/blob/master/src/storage.ts#L12